### ISSUE TITLE
Deleting objects not conforming to a predicate

### ISSUE THREAD

**Issue by tspecht** [realm/realm-cocoa#1115]
Hey guys,

I'm about to delete quite some objects from my realm. Precisely I want to delete all objects which IDs (which is the primary key of the class) are not in a given array. Therefore I'm performing the following call:

[[RLMRealm defaultRealm] deleteObjects:[DSSnip objectsWithPredicate:[NSPredicate predicateWithFormat:@"NOT(id IN %@) AND isLocal = NO", snipIds]]]

Unfortunately, the snipIds array can become quite huge (> 1000 entries). I do not know if it is for that reason but the performance on this call is quite bad. I could offload it to a background thread but I need to do writer on the same realm and therefore this long call is blocking my other transactions. Any ideas on how to improve performance?

**Comment by ppaulojr** [comment]
I'm using Realm 2.2.0 and I'm having the exact same problem described above. It seems that making a query with `NOT IN` and using the results of it is very slow.

**Comment by kunalsood** [comment]
@ppaulojr Unfortunately, this was never resolved. I resorted to (and still am doing so) converting my RLMResults object to an NSSet using something like `[NSSet setWithArray:[resultsObject valueForKey:@"self"]];`, and then applying the `NOT IN` predicate to this set using `filteredSetUsingPredicate:` which appears to be fast/efficient enough for majority of my use cases.

Either way, you might have better luck getting attention on this issue by opening a new issue, as this issue has been closed for over 2 years now.
